mesheryctl/internal/cli/root/system/start.go

```diff
@@ -105,13 +105,18 @@ func start() error {
 		return err
 	}
 
-	if utils.PlatformFlag == "docker" || utils.PlatformFlag == "kubernetes" {
-		currCtx.SetPlatform(utils.PlatformFlag)
-	} else {
-		if utils.CfgFile != utils.DefaultConfigPath {
+	if utils.PlatformFlag == "" {
+		utils.PlatformFlag = currCtx.GetPlatform()
+	}
+	if utils.PlatformFlag != "" {
+		if utils.PlatformFlag == "docker" || utils.PlatformFlag == "kubernetes" {
+			currCtx.SetPlatform(utils.PlatformFlag)
+		} else {
+			if utils.CfgFile != utils.DefaultConfigPath {
+				return fmt.Errorf("the platform '%s' is not supported. Supported platforms are:\n\n- docker\n- kubernetes\n\nVerify this setting in your meshconfig at %s or verify by executing `mesheryctl system context view`", utils.PlatformFlag, utils.CfgFile)
+			}
 			return fmt.Errorf("the platform '%s' is not supported. Supported platforms are:\n\n- docker\n- kubernetes\n\nVerify this setting in your meshconfig at %s or verify by executing `mesheryctl system context view`", utils.PlatformFlag, utils.CfgFile)
 		}
-		return fmt.Errorf("the platform '%s' is not supported. Supported platforms are:\n\n- docker\n- kubernetes\n\nVerify this setting in your meshconfig at %s or verify by executing `mesheryctl system context view`", utils.PlatformFlag, utils.CfgFile)
 	}
```

Comment:
we still need to handle the case of utils.PlatformFlag == "" case, right?

Reply:
actually we are reversing the case we check if the flag's value != "". then we can check if it is docker or kubernetes and if it is none of them we return err.

Reply:
and default case for the flag value would be "" empty string in that case we will fetch from the config file.